pool/src/transaction_pool.rs

```diff
@@ -171,9 +177,37 @@ impl TransactionPool {
 			self.adapter.tx_accepted(&entry.tx);
 		}
 
+		// Transaction passed all the checks but we have to make space for it
+		if evict {
+			self.evict_from_txpool()?;
+		}
+
 		Ok(())
 	}
 
+	// Remove the last transaction from the flattened bucket transactions.
+	// No other tx depends on it, it has low fee_to_weight and is unlikely to participate in any cut-through.
+	pub fn evict_from_txpool(&mut self) -> Result<(), PoolError> {
+		// Get bucket transactions
+		let bucket_transactions = self.txpool.bucket_transactions(Weighting::NoLimit);
+
+		// Get last transaction and remove it
+		match bucket_transactions.last() {
+			Some(evictable_transaction) => {
+				// Remove transaction
+				self.txpool.entries = self
+					.txpool
+					.entries
+					.iter()
+					.filter(|x| x.tx != *evictable_transaction)
+					.map(|x| x.clone())
+					.collect::<Vec<_>>();
+				Ok(())
+			}
+			None => Err(PoolError::OverCapacity),
```

Comment:
This does not make sense - if we ever call `evict_from_pool` then we should be reasonably confident we have at least one tx in the pool.
But if entries was to ever be empty we should probably just do nothing and return (and not raise an error).

Reply:
Indeed fixed.